### PR TITLE
user-interface: raise test coverage from 72.22% to 85.73%

### PR DESCRIPTION
**Pull request by brandonkindred**
## Summary

- Adds tests for **every previously-untested API service** (14 services) and extends existing service specs to cover all public methods, including SSE streams, polling loops, error paths, and edge cases
- Adds focused component tests for ~20 components (accessibility audit form, accessibility findings, accessibility report, accessibility design system, all 7 PA personal assistant components, persona test audit panel, architecture results, investment proposal/promotion/profile-form/chat, run-team-form, backend/frontend code v2 forms, social marketing run form, pending questions, question card, team-assistant-chat, branding-chat) and the road-trip-planning slot-filler utility module
- Extends the HTTP error interceptor spec to cover every status branch (400/401/403/404/418/500/503/0) and both shapes of the validation detail payload
- Overall line coverage moves from **72.22% → 85.73%** (15.5k → 16.1k lines hit out of 18,785 total in `src/app/**/*.ts`); 1228+ tests pass; `npm run lint` is clean

## Coverage delta — selected highlights

| File / area | Before | After |
|---|---:|---:|
| **Overall** (src/app/**/*.ts lines) | 72.22% | **85.73%** |
| `services/deepthought-api.service.ts` | 7.7% | ~95% |
| `services/software-engineering-api.service.ts` | 24.2% | ~95% |
| `services/investment-api.service.ts` | 34.6% | ~95% |
| `services/blogging-api.service.ts` | 37.4% | ~95% |
| `services/branding-api.service.ts` | 48.7% | ~95% |
| `services/personal-assistant-api.service.ts` | 60.7% | ~95% |
| `services/job-actions.service.ts` | 47.1% | 100% |
| `services/nav-state.service.ts` | 45.5% | 100% |
| `services/sales-api.service.ts` | 43.5% | 100% |
| `services/team-assistant-api.service.ts` | 40.6% | 100% |
| `services/nutrition-api.service.ts` | (no spec) | ~95% |
| `services/road-trip-planning-api.service.ts` | (no spec) | 100% |
| `services/startup-advisor-api.service.ts` | (no spec) | ~95% |
| `services/agentic-team-api.service.ts` | (no spec) | 100% |
| `components/road-trip-planning-dashboard/trip-slot-filler.ts` | 13.3% | ~99% |
| `components/accessibility-audit-form/...` | 43.9% | ~99% |
| `components/accessibility-findings/...` | 57.8% | ~99% |
| `components/accessibility-report/...` | 43.1% | ~99% |
| `components/accessibility-design-system/...` | 52.0% | ~99% |
| `components/pa-tasks/...` | 54.8% | ~99% |
| `components/pa-calendar/...` | 40.7% | ~99% |
| `components/pa-chat/...` | 66.0% | ~99% |
| `components/pa-deals/...` | 52.3% | ~99% |
| `components/pa-documents/...` | 62.6% | ~99% |
| `components/pa-profile/...` | 59.2% | ~99% |
| `components/pa-reservations/...` | 47.9% | ~99% |
| `components/investment-proposal/...` | 45.2% | ~99% |
| `components/investment-promotion/...` | 61.1% | ~99% |
| `components/investment-profile-form/...` | 59.2% | ~99% |
| `components/investment-chat/...` | 47.4% | ~99% |
| `components/architecture-results/...` | 46.8% | 100% |
| `components/persona-test-audit-panel/...` | 46.1% | ~99% |
| `components/pending-questions/...` | 62.3% | ~95% |
| `components/pending-questions/question-card/...` | 71.1% | 100% |
| `components/team-assistant-chat/...` | 42.7% | ~95% |
| `components/branding-chat/...` | 54.9% | ~95% |
| `components/run-team-form/...` | 82.7% | ~99% |
| `components/backend-code-v2-run-form/...` | 64.9% | 100% |
| `components/frontend-code-v2-run-form/...` | 64.9% | 100% |
| `components/social-marketing-run-form/...` | 66.1% | ~99% |
| `core/error-handler.interceptor.ts` | 65.5% | ~99% |

## Files still below 95% (largest gaps)

These are all large dashboard components (300-900 lines each) that wrap multi-tab workflows, SSE streams, and complex state — extending their specs to 95% would be sizeable follow-up PRs:

| File | Current | Uncovered lines |
|---|---:|---:|
| `components/run-team-tracking/run-team-tracking.component.ts` | 39.1% | 558 |
| `components/blogging-dashboard/blogging-dashboard.component.ts` | 29.1% | 487 |
| `components/branding-dashboard/branding-dashboard.component.ts` | 60.8% | 272 |
| `components/integrations-dashboard/integrations-dashboard.component.ts` | 47.7% | 237 |
| `components/jobs-dashboard/jobs-dashboard.component.ts` | 75.4% | 187 |
| `components/deepthought-dashboard/deepthought-dashboard.component.ts` | 46.2% | 140 |
| `components/ai-systems-dashboard/ai-systems-dashboard.component.ts` | 59.3% | 105 |
| `components/agent-provisioning-dashboard/agent-provisioning-dashboard.component.ts` | 59.6% | 103 |
| `components/investment-strategy/investment-strategy.component.ts` | 85.5% | 58 |
| `components/persona-testing-dashboard/persona-testing-dashboard.component.ts` | 77.1% | 52 |
| `components/app-shell/app-shell.component.ts` | 64.2% | 38 |
| (~30 other components/services with smaller gaps) | various | ≤30 each |

No production code was modified. No `/* istanbul ignore */` directives were added — every covered branch is exercised by real behavioral assertions.

## Test plan

- [x] `cd user-interface && npm ci`
- [x] `npm run test:coverage` — 1228+ tests pass, overall 85.73% line coverage
- [x] `npm run lint` — clean
- [ ] CI runs the full coverage gate (note: the project's hard 95% target is not yet met — remaining work is concentrated in the 8 dashboards listed above)

https://claude.ai/code/session_01V5u4zW36taPZEx99n69c4n

---
_Generated by [Claude Code](https://claude.ai/code/session_01V5u4zW36taPZEx99n69c4n)_

### CODE REVIEW

**Comment by brandonkindred** [comment]
@codex review

File: user-interface/src/app/components/ai-systems-dashboard/ai-systems-dashboard-extra.spec.ts [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+      listBlueprints: vi.fn().mockReturnValue(of({ blueprints: ['bp1', 'bp2'] })),
+      getBlueprint: vi.fn().mockReturnValue(of({ project_name: 'bp1' })),
+      getJobStatus: vi.fn().mockReturnValue(of({ job_id: 'j1', status: 'completed' })),
+      cancelJob: vi.fn().mockReturnValue(of({})),
+      deleteJob: vi.fn().mockReturnValue(of({})),
+    };
+    await TestBed.configureTestingModule({
+      imports: [AISystemsDashboardComponent, NoopAnimationsModule],
+      providers: [
+        provideHttpClient(),
```
[the remark this answers is not in the document]
Comment (brandonkindred) [reply user-interface/src/app/components/ai-systems-dashboard/ai-systems-dashboard-extra.spec.ts:43]: Fixed in `127af85`. Added `provideHttpClientTesting()` from `@angular/common/http/testing` alongside the existing `provideHttpClient()`, which installs an `HttpTestingController` that intercepts every HTTP call (including the nested `TeamAssistantChatComponent.ngOnInit → loadConversation()` requests) — no more real network I/O. Verified the spec passes 25/25 and the full directory run is 29/29 with no `AggregateError`, fetch-failure, or unhandled-rejection traces in the output.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by brandonkindred** [comment]
@codex review
